src/lib/components/home/artwork_rails/artwork_rail.tsx

```diff
@@ -207,13 +207,21 @@ class ArtworkRail extends React.Component<Props & RelayPropsWorkaround, State> {
     if (this.state.loadFailed) {
       return null
     }
+    const hasFetched = this.props.relay && this.props.relay.variables.fetchContent
 
     const sideMargin = isPad ? 40 : 20
     const style: any = { marginLeft: sideMargin, marginRight: sideMargin }
-    if (!(this.props.relay && this.props.relay.variables.fetchContent)) {
+    if (!hasFetched) {
       // if there's no content, set minHeight to prevent spinners from all the rails showing
       style.minHeight = 400
     }
+
+    const hasArtworks = hasFetched && this.props.rail.results && this.props.rail.results.length
+    if (!hasArtworks) {
```

Comment:
Isn’t this going to return `null` while it’s still fetching and thus not show the spinner like it should?